Mathlib/Algebra/BigOperators/Expect.lean

```diff
@@ -0,0 +1,403 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Algebra.Algebra.Rat
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Data.Finset.Pointwise.Basic
+import Mathlib.Data.Fintype.BigOperators
+import Mathlib.GroupTheory.GroupAction.BigOperators
+
+/-!
+# Average over a finset
+
+This file defines `Finset.expect`, the average (aka expectation) of a function over a finset.
+
+## Notation
+
+* `𝔼 i ∈ s, f i` is notation for `Finset.expect s f`. It is the expectation of `f i` where `i`
+  ranges over the finite set `s` (either a `Finset` or a `Set` with a `Fintype` instance).
+* `𝔼 i, f i` is notation for `Finset.expect Finset.univ f`. It is the expectation of `f i` where `i`
+  ranges over the finite domain of `f`.
+* `𝔼 i ∈ s with p i, f i` is notation for `Finset.expect (Finset.filter p s) f`. This is referred to
+  as `expectWith` in lemma names.
+* `𝔼 (i ∈ s) (j ∈ t), f i j` is notation for `Finset.expect (s ×ˢ t) (fun ⟨i, j⟩ ↦ f i j)`.
+
+## TODO
+
+Connect `Finset.expect` with the expectation over `s` in the probability theory sense.
+-/
```

Comment:
imo we should add an implementation note here explaining why this definition is helpful to have, as it isn't immediately clear otherwise. That is, why is this useful when probability expectation exists, and why is it not just an abbreviation on top of Finset.sum. Of course I'm fully on board with this definition being in mathlib, but to future readers / contributors these factors may not be clear.

Reply:
I'm not sure I addressed exactly your point. Please comment if you want further modifications